jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -323,18 +327,21 @@ private final long memSegmtOfPtrToLongArg(MemorySegment argValue, LinkerOptions
 			 * is captured so as to reset the internal index and avoid retaining the references
 			 * to the unreachable objects.
 			 */
-			if (info != null) {
-				info.clear();
+			if(!flagCreateHeapArgInfo.get() && !infoStack.isEmpty()) {
```

Comment:
The second check `!infoStack.isEmpty()` is not needed. If flagCreateHeapArgInfo is false there should be a stack entry. Also add a space before the first parenthesis.